### PR TITLE
Added ONNX end-to-end tests

### PR DESCRIPTION
**Pull request by maartenvds**
Added ONNX end-to-end tests
* Added `test_data` folder recursively with git lfs
* Also removed obsolete .gitlab-ci.yml

### CODE REVIEW

File: diana/byoc/test.py [hunk 1]
```diff
@@ -136,3 +142,47 @@ def test_mlperf_tiny_mobilenet(run, tmp_path):
...
+    d.tvm_compile()
+    d.gcc_compile(gcc_opt=0)
+    d.run(gdb=False)   # raises exception on failure
+
+
+def test_mlperf_tiny_ds_cnn_onnx_x86(run, tmp_path):
+    run_onnx_x86('test_data/export_dscnn/DSCNN_QL_NOANNOTATION.onnx', tmp_path)
+
+
+def test_mlperf_tiny_resnet_onnx_x86(run, tmp_path):
```
Comment (JosseVanDelm) [inline diana/byoc/test.py:171]: Can you either:
* Add a conditional skip to the following tests if git lfs is not used? And mention in the repo that LFS is required for this type of tests
* Add git lfs to the requirements in the README.md? And require them for all tests

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Hi Maarten, this is great!
However, I'm a bit concerned that It might not be fully clear that we are using LFS for this.
So could you make it clear which tests require LFS and which not?
Besides that, it's also a bit weird that the new tests accept the run argument but don't use it, but okay if this is solved in the future when you get hardware access.

File: diana/byoc/test.py [hunk 2]
```diff
@@ -136,3 +142,47 @@ def test_mlperf_tiny_mobilenet(run, tmp_path):
...
+    assert os.path.exists(os.path.join(onnx_dir, 'output.npy'))
+
+    mod, params = driver.parse_onnx_input(onnx_file)
+    d = driver.X86Driver(mod, params, directory)
+    d.tvm_compile()
+    d.gcc_compile(gcc_opt=0)
+    d.run(gdb=False)   # raises exception on failure
+
+
+def test_mlperf_tiny_ds_cnn_onnx_x86(run, tmp_path):
```
[the remark this answers is not in the document]
Comment (maartenvds) [reply diana/byoc/test.py:167]: Maybe for these `_x86` tests it makes more sense that we remove the `run` argument here. Later, we make new `_pulp` variants of these tests with a `run` argument: if no target is connected, we only compile, with target we can run them on DIANA. What do you think of my proposal?

File: diana/byoc/test.py (hunk 1 above)
Comment (maartenvds) [reply diana/byoc/test.py:171]: I'll update the README.md that they have to `git lfs pull` before running `pytest -v test.py`.

**Review by JosseVanDelm — APPROVED** [review]
Minor nits in the README but otherwise looks good to me!

**Pull request merged**